compiler/modules/types/list.bal

```diff
@@ -320,8 +317,14 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, ListC
             }
         }
         SemType rd = diff(rest, nt.rest);
-        if !isEmpty(cx, rd) && listInhabited(cx, members, rd, neg.next) {
-            return true;
+        if !isEmpty(cx, rd) {
+            // We have checked the posibilities of existance of a shape in list with fixedLength from 0 to maxInitialLen (exclusive).
+            // Check existance of a shape with more than `maxInitialLen` numuber of members.
+            FixedLengthArray s = fixedArrayShallowCopy(members);
+            fixedArrayFill(s, int:max(len, maxInitialLen), rest);
```

Comment:
This is correct, but I think it would be less confusing to fill only of maxInitialLen > len.